flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartitionView.java

```diff
@@ -39,6 +40,8 @@
 	/** Flag indicating whether this view has been released. */
 	private final AtomicBoolean isReleased;
 
+	private volatile boolean isBlockedByCheckpoint = false;
```

Comment:
It is sensitive to maintain a volatile variable in core stack. Actually we can avoid it for `RemoteInputChannel` by accessing it only by netty stack, it would only bring one more unnecessary `UserEventTrigger` in netty stack. But for local channel, it has problems now, let us think whether they are other options to bypass.

Reply:
As discussed, I moved the blockedByCheckpoint tag to subpartition.